### PR TITLE
v5.17.2

### PR DESCRIPTION
**Pull request by flaviendelangle**

### CODE REVIEW

File: CHANGELOG.md
Comment (flaviendelangle) [inline CHANGELOG.md:18]: For v6 we should always release all our stable packages with the same version, but I think that for 1 patch before the alpha it's not worth jumping the pickers to 5.17.2

File: CHANGELOG.md
Comment (flaviendelangle) [inline CHANGELOG.md:16]: @alexfauquette do you think this one is worth highlighting ?

File: packages/x-date-pickers-pro/package.json
Comment (LukasTy) [inline packages/x-date-pickers-pro/package.json:51]: We should align this dependency version on all packages. 🤔

File: packages/storybook/package.json
Comment (LukasTy) [inline packages/storybook/package.json:25]: Missing `grid` and `grid-pro` bumps.
```suggestion
    "@mui/x-data-grid": "5.17.2",
    "@mui/x-data-grid-generator": "5.17.2",
    "@mui/x-data-grid-premium": "5.17.2",
    "@mui/x-data-grid-pro": "5.17.2",
    "@mui/x-license-pro": "5.17.2",
```

Comment (flaviendelangle) [reply packages/storybook/package.json:25]: We are not releasing packages with no change
Not sure if it's a good strategy, but it's how our Lerna is configured

File: packages/x-date-pickers-pro/package.json
Comment (flaviendelangle) [reply packages/x-date-pickers-pro/package.json:51]: https://github.com/mui/mui-x/pull/6086#discussion_r965988533

I was discussing it here
Theoretically I agree, but I was wondering if for just 1 patch we could avoid adding a weird version bump to our users

File: packages/storybook/package.json
Comment (LukasTy) [reply packages/storybook/package.json:25]: But in changelog we are listing all the packages 🤔 
Don't we have a decision to try and maintain same versions for all the packages to reduce the headache for users?

File: packages/x-date-pickers-pro/package.json
Comment (LukasTy) [reply packages/x-date-pickers-pro/package.json:51]: I'm pretty sure that with previous releases we always bumped `license` package version to the same one as the whole package in all the sub-packages. 🤔 
Correct me if I'm wrong..?

File: packages/storybook/package.json
Comment (flaviendelangle) [reply packages/storybook/package.json:25]: I think the comes from the core where some package can go without updates for months and it would cause a lot of "useless" versions.

But to be honest I am not very up to date with this topic.
It almost never occurred before, except for the license package

Maybe @mnajdova, @DanailH or @oliviertassinari will know better.

File: packages/x-date-pickers-pro/package.json
Comment (flaviendelangle) [reply packages/x-date-pickers-pro/package.json:51]: Not if you use `year release:version` at least
There is no 5.17.1 for instance https://www.npmjs.com/package/@mui/x-license-pro

Comment (flaviendelangle) [reply packages/x-date-pickers-pro/package.json:51]: @LukasTy you answered in the wrong thread I think

File: CHANGELOG.md
Comment (flaviendelangle) [reply CHANGELOG.md:20]: No strong opinion here
I just think bumping from v5.0.0 to v5.17.2 to v6.0.0-alpha.1 can be confusing.
We probably should have bumped to v5.17.0 instead of 5.0.0 last week.

Comment (LukasTy) [reply CHANGELOG.md:20]: > We probably should have bumped to v5.17.0 instead of 5.0.0 last week.

Yes, that's my thinking as well, if we did not align the version with the "stable" release - no point in doing that now.

File: packages/x-date-pickers-pro/package.json
Comment (LukasTy) [reply packages/x-date-pickers-pro/package.json:51]: I'm pretty sure I answered in the correct thread.
Anyways, I'm all for releasing packages once it makes sense. 😉 
Was initially thinking that some other package had a difference license package version, but if they all match - fine by me. 👌 👍

File: CHANGELOG.md
Comment (alexfauquette) [reply CHANGELOG.md:18]: Since we have nothing else, why not

File: CHANGELOG.md
Comment (alexfauquette) [inline CHANGELOG.md:14]: I assume that's what @LukasTy  is proposing to be consistent with the unmodified community and pro

```suggestion
### `@mui/x-data-grid-premium@v5.17.2`
```

File: packages/storybook/package.json
Comment (alexfauquette) [reply packages/storybook/package.json:25]: +1 for not releasing `@mui/x-data-grid v5.17.2` and release `@mui/x-data-grid v5.17.3` next time if necessary

**Review by alexfauquette — APPROVED** [review]

File: CHANGELOG.md
Comment (flaviendelangle) [reply CHANGELOG.md:15]: Matheus pushed a fix on the Community package so the discussion is closed I guess
But I agree :+1:

**Review by m4theushw — APPROVED** [review]
If you want you can include https://github.com/mui/mui-x/pull/6030. It only needs a sync with `master` to fix argos.

**Comment by LukasTy** [comment]
> If you want you can include #6030. It only needs a sync with `master` to fix argos.

👍 Synced and merged it to master 🙈

**Comment by flaviendelangle** [comment]
Added here

**Review by LukasTy — APPROVED** [review]

**Comment by DanailH** [comment]
Since this is the last planned release for v5 should we merge this PR https://github.com/mui/mui-x/pull/5744 to have it in the docs? It looks like it has the needed approvals it's only that @cherniavskii is on PTO?

**Review by DanailH — APPROVED** [review]

**Comment by joserodolfofreitas** [comment]
I'd like to merge the main demo too

**Comment by flaviendelangle** [comment]
@joserodolfofreitas @DanailH I added a note for the v6, if you can review it.

For the main demo, it's quite independent, if you want we can merge it next week on `master` an release a patch

File: CHANGELOG.md
Comment (DanailH) [inline CHANGELOG.md:12]:
```suggestion
The `v5.x` will now only receive critical bug fixes.
```

File: CHANGELOG.md
Comment (joserodolfofreitas) [inline CHANGELOG.md:11]:
```suggestion
From now on, we'll be focusing on developing MUI X v6. 
You can check the [roadmap](https://github.com/mui/mui-x/projects/1) for more details on what's coming next.

And if you'd like to help, please consider volunteering to give us a [user interview](https://docs.google.com/forms/d/11L_zxL7oD_B-ZrZDuSyIkUzJLzOTUU2M4vHXxMVtWhU/edit).
We'd love to know more about your use cases, pain points and expectations for the future.
```

File: CHANGELOG.md
Comment (joserodolfofreitas) [inline CHANGELOG.md:12]:
```suggestion
The `v5` packages will only get new versions to patch critical bug fixes.
```

**Comment by joserodolfofreitas** [comment]
The [PR is there](https://github.com/mui/mui-x/pull/5292) for a couple of months, and basically everyone commented on it already. Any improvements can go on a new PR (plus, it's not a components' feature)

**Comment by flaviendelangle** [comment]
> The https://github.com/mui/mui-x/pull/5292 for a couple of months, and basically everyone commented on it already. Any improvements can go on a new PR (plus, it's not a components' feature)

We were not very reactive on its review I agree
I still think rushing the merge right now is not a good idea

**Review by joserodolfofreitas — APPROVED** [review]

File: CHANGELOG.md
Comment (joserodolfofreitas) [inline CHANGELOG.md:14]:
```suggestion
And if you'd like to help, please consider volunteering to give us a [user interview]([contacting us](https://forms.gle/vsBv6CLPz9h57xg8A)).
```

File: CHANGELOG.md
Comment (oliviertassinari) [reply CHANGELOG.md:31]: I was about to leave a new comment until I found this thread. I agree with the conclusion 👍. If we need to release more versions, it will be clearer with a `@mui/x-date-pickers-pro@v5.17.2`. So it can look like this in people's `package.json`:

```json
  "@mui/x-data-grid": "5.17.2",
  "@mui/x-date-pickers": "5.17.2",
```

File: packages/storybook/package.json
Comment (flaviendelangle) [reply packages/storybook/package.json]: I think there is a real question for packages of the same "family"

Skipping releases on `@mui/x-license-pro` is one thing
Skipping releases on `@mui/x-data-grid` / `@mui/x-data-grid-pro` when we change only `@mui/x-data-grid-premium` is a little weird to me

**Pull request merged**
